Add ancestor array

scottmessinger opened the pull request:
One of the ideas I've toyed around with is to have a `api/standards` endpoint that would have all the information to display a standard. I don't know if it's useful to anyone so I haven't done it. If we were to do it, I imagine it would look something like what we send to Algolia for search. It wouldn't make much sense to send this back in an API response for a standard in the `standard_set` response as it would dramatically increase the size of the response. However, does it make sense to include it as a separate endpoint? 

Specifically, @aleksclark, I'm wondering if caching the hierarchy (along with other data on the standard) would help solve your pain point with the current API.

Note: I changed the code in the `send_to_algolia` file. If we did this, it would not still live in this file or this class.

scottmessinger committed:
Reorganize scripts and add support for:

- parsing standards document
- finding standard set queries
- giving standards hierarhcy

scottmessinger committed:
Fix asn resource parser and refactor key matchers

scottmessinger committed:
First pass at a working API

scottmessinger committed:
Correctly order standards

scottmessinger committed:
Added cc exceptions and renamed education level

scottmessinger committed:
Import Standard Sets

scottmessinger committed:
Return standard sets from the API

scottmessinger committed:
Clean up files in preparation to share

scottmessinger committed:
cleanup cli files

scottmessinger committed:
Dump jurisdictions collection

scottmessinger committed:
Moved mongodb dump

scottmessinger committed:
add more auth

scottmessinger committed:
Move some files and add support for commits

scottmessinger committed:
add swagger

scottmessinger committed:
Add algolia search

scottmessinger committed:
Rename standardsSet to standardSet

scottmessinger committed:
Validate tokens

scottmessinger committed:
Made script to cache for algolia

scottmessinger committed:
Added back important bit of code that I accidentally took out earlier

scottmessinger committed:
Updated list of ASN documents

scottmessinger committed:
Polished importer, moved files around, renamed standards_documents

Renamed standards_documents to be standard_documents to conform to
normal english conventions around plurals

scottmessinger committed:
Save StandardSet version & send to Algolia & rename

scottmessinger committed:
Make all versions of standard document have the same id

scottmessinger committed:
Update path

scottmessinger committed:
Rename send_to_algolia

scottmessinger committed:
Clean up mongo init and move things around

scottmessinger committed:
Set ruby version in gemfile

scottmessinger committed:
add patch level to ruby version

scottmessinger committed:
Add puma

scottmessinger committed:
Clean up names and add sitemap

scottmessinger committed:
Reorganized files

scottmessinger committed:
move send to algolia

scottmessinger committed:
Make source to subject mapping more clear

scottmessinger committed:
Clean up scripts to generate jurisdiction mapping

scottmessinger committed:
Add comments

scottmessinger committed:
Change shape of jurisdiction

scottmessinger committed:
import all the standards

scottmessinger committed:
Add subject back in

scottmessinger committed:
Fix subject on importer

scottmessinger committed:
Add comments

scottmessinger committed:
Add license

scottmessinger committed:
Remove unnecessary file

scottmessinger committed:
Add ancestor array

aleksclark commented:
I think that a `api/standards` endpoint makes sense, and furthermore we should probably allow requests for multiple standard ids. Standards tagging is going to tend to be a HABTM relationship, so it will be useful to be able to return all the tag data for a given entity. For example, I tag a lesson with CC, Texas, and District standards, and when the UI needs to display, it hits the api with `api/standards?id[]=123&id[]=456&id[]=789`

scottmessinger committed:
Set logger level to debug

scottmessinger committed:
Change max/min pool size

scottmessinger committed:
Change pool size

scottmessinger committed:
remove pool size

scottmessinger committed:
Add rack defalater

scottmessinger committed:
Deflate all the responses

scottmessinger committed:
Change endpoint for generating standard set from query

scottmessinger committed:
Add skylight

scottmessinger committed:
Add comments

scottmessinger committed:
Add skylight

scottmessinger committed:
One more attempt at skylight

scottmessinger committed:
Remove all traces of skylight

scottmessinger committed:
Fully remove skylight

scottmessinger committed:
Add ancestor_id to the standard response

scottmessinger committed:
whitespace

scottmessinger committed:
Whitespace (to trigger heroku)

scottmessinger committed:
More whitespace to trigger heroku

scottmessinger committed:
Whitespace to trigger heroku build

scottmessinger committed:
Add a sitemap.xml endpoint

As I note in the comment, this isn’t ideal. If I take the time to
customize the routing on the fronted, I’ll gladly move this to the
sinatra main.rb file.

scottmessinger committed:
Add commits/reject endpoint

scottmessinger committed:
Correctly show commits

scottmessinger committed:
Updated the API

scottmessinger committed:
Change readme to MD

scottmessinger committed:
Typo in readme

scottmessinger committed:
Update copyright on license

scottmessinger committed:
Upgrade mongo

scottmessinger committed:
Add traceview

scottmessinger committed:
add trace view to api

scottmessinger committed:
More trace view stuff

scottmessinger committed:
Fix traceview

scottmessinger committed:
Remove traceview

scottmessinger committed:
Add new relic

scottmessinger committed:
require newrelic

aleksclark committed:
add foreman to the gemfile for local dev, add editorconfig, update readme with instructions

aleksclark committed:
change mongo db name to match exported jurisdiction dump.

aleksclark committed:
wrap up readme instructions

scottmessinger committed:
Merge pull request #2 from aleksclark/dev-tweaks

add foreman to the gemfile for local dev, add editorconfig, update README

scottmessinger committed:
Fix problem with standards with depth 0 having ancestors

scottmessinger committed:
Make sitemap.xml a hidden endpoint in the swagger docs

scottmessinger committed:
Merge branch 'master' of https://github.com/commonstandardsproject/api

scottmessinger committed:
Add link to mailing list

scottmessinger committed:
Add setup script and change readme

scottmessinger committed:
Merge with master

scottmessinger committed:
Add a version to the API

scottmessinger committed:
Ignore .log files

scottmessinger committed:
Update README.md

scottmessinger committed:
Add link to swagger docs

scottmessinger committed:
Add support for creating standard sets

scottmessinger committed:
Ignore .log files

scottmessinger committed:
Merge branch 'master' of https://github.com/commonstandardsproject/api

scottmessinger committed:
Upgrade skylight & grape

scottmessinger committed:
Add skylight.yml

scottmessinger committed:
typo

scottmessinger committed:
Add skylight

scottmessinger committed:
Accept commits

scottmessinger committed:
Reject standards that are ""

scottmessinger committed:
Change to work with new diff/ops format

scottmessinger committed:
Ignore new relic

scottmessinger committed:
Ignore if commit is applied

scottmessinger committed:
Clean up .gitignore

scottmessinger committed:
Save standard set subject on commit

scottmessinger committed:
disable new relic in dev

scottmessinger committed:
Upgrade grape swagger and hide endpoint

scottmessinger committed:
Update with new standards data

scottmessinger committed:
Don't print denormalize

scottmessinger committed:
Add bson_ext

scottmessinger committed:
Send to slack on commit submission

jfeldstein committed:
Add .ruby-gemset file to compliment .ruby-version

jfeldstein committed:
Hash keys are strings

Fixes issue where:

$ ./importer/cli setup
D, [2015-12-19T17:37:11.928138 #26864] DEBUG -- : MONGODB | Adding 127.0.0.1:27017 to the cluster. | runtime: 0.0079ms
D, [2015-12-19T17:37:11.929493 #26864] DEBUG -- : MONGODB | COMMAND | namespace=admin.$cmd selector={:ismaster=>1} flags=[] limit=-1 skip=0 project=nil | runtime: 1.2133ms
./importer/cli:20:in `setup': undefined local variable or method `apiKey' for #<ImporterCLI:0x007fad7a041030> (NameError)
	from /Users/jordan/.rvm/gems/ruby-2.2.2@common-standards-project/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/jordan/.rvm/gems/ruby-2.2.2@common-standards-project/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/jordan/.rvm/gems/ruby-2.2.2@common-standards-project/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /Users/jordan/.rvm/gems/ruby-2.2.2@common-standards-project/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from ./importer/cli:46:in `<main>'

scottmessinger committed:
Add basic pull request model

scottmessinger committed:
Merge pull request #5 from jfeldstein/string-keys

Hash keys are strings

scottmessinger committed:
Merge pull request #4 from jfeldstein/with-gemset

Add .ruby-gemset file to compliment .ruby-version

scottmessinger committed:
WIP Add new pull request models and endpoints

scottmessinger committed:
Find all

scottmessinger committed:
Add first integration with postmark client

scottmessinger committed:
Add email and asana notifications

scottmessinger committed:
Add test for the pull requests

scottmessinger committed:
Add test and code to add comments

scottmessinger committed:
Fix various bugs

scottmessinger committed:
Fix sending to asana and guard input

scottmessinger committed:
Update title on Asana

scottmessinger committed:
Approve jurisdictions and upgrade algolia

scottmessinger committed:
Update Email

scottmessinger committed:
Don't check origins for csp.com

scottmessinger committed:
Default submitter email

scottmessinger committed:
Don't validate submitter email

scottmessinger committed:
Remove commit entity

scottmessinger committed:
Add support for creating PRs from commits

scottmessinger committed:
Merge branch 'master' of https://github.com/commonstandardsproject/api

scottmessinger committed:
Use new ops format

scottmessinger committed:
save the listId

scottmessinger committed:
default document to a {}

scottmessinger committed:
Add new standard sets and make more resilient

scottmessinger committed:
Update readme as you don't have to import jurisdictions anymore

scottmessinger committed:
Remove old documents

scottmessinger committed:
Merge branch 'pull-request'

scottmessinger committed:
Fix names of methods in CLI

scottmessinger committed:
add back script to import

scottmessinger committed:
Support BC educational levels

scottmessinger committed:
Match any Canadian province educational levels

scottmessinger committed:
Remove import doc

scottmessinger committed:
Add note about the standards importer

scottmessinger committed:
return id with standard set

scottmessinger committed:
return id with standard set

scottmessinger committed:
Add new jurisdictions and matchers

scottmessinger committed:
Add new standards -- July 27, 2016

scottmessinger committed:
Merge branch 'master' of https://git.heroku.com/common-standards-api

scottmessinger committed:
Ensure version is created

scottmessinger committed:
Add picture of homepage
